pkg/config/utils/endpoints_test.go

```diff
@@ -34,6 +38,14 @@ func TestSecretBackendWithMultipleEndpoints(t *testing.T) {
 	assert.Equal(t, expectedKeysPerDomain, keysPerDomain)
 }
 
+func getMockConfig(t *testing.T, yamlData string) model.Config {
```

Comment:
How do you feel about moving this function to `pkg/config/mock/helper/read_mock_config.go`, renaming it to `ReadMockConfig(t *testing.T, yamlData string) model.Config` and using it for `comp/rdnsquerier/impl/config_test.go` while we're here?

I think it's good that this PR is removing the redundant, duplicated one-off helper methods to read and create the config. But personally, I think it's okay to have a single helper method in one obvious place that many tests in various packages can depend on and reuse. Otherwise, authors will be incentivized to re-create (or copy-paste) the same utilities throughout the code in order to fill the same need.